bintrace.c

```diff
@@ -219,7 +265,10 @@ static void append_trace_block()
 
 	if (new_block == MAP_FAILED) {
 		perror("Failed to append new trace block");
-		dump_hist();
+		struct flags fail_flags;
```

Comment:
Why do we disable those when dumping because of error?  
(btw, I think It's ok to modify global options here if we call exit anyway)